website/common/script/fns/randomDrop.js

```diff
@@ -114,8 +126,14 @@ export default function randomDrop (user, options, req = {}, analytics) {
         randomVal(pickBy(content.hatchingPotions, (v, k) => acceptableDrops.indexOf(k) >= 0)),
       );
 
-      user.items.hatchingPotions[drop.key] = user.items.hatchingPotions[drop.key] || 0;
-      user.items.hatchingPotions[drop.key] += 1;
+      user.items.hatchingPotions = {
```

Comment:
only the second one here is necessary to trigger Vue reactivity

Reply:
Are you sure that not the first? The second one already exist and have setter to trigger reactivity

Reply:
I mean, if it's the second or the first doesn't really matter, the important thing is that the whole object is reassigned at least once with the new property (in case [drop.key] did not exist and is set to 0). Does it make sense?

In the old line 117 the `drop.key` property could be missing (notice the `|| 0`) so we want to reassign the whole object to trigger reactivity in case the property was missing. If you do it on the first assignment or the second is the same because at the end the entire object has been reassigned with the new property having the correct getter and setter

Reply:
Sure, you are right. So I set it to first, due to rule "if key is not exist, we should reassign whole parent".